### PR TITLE
chore: update migration handling for provider consensus v2->v3

### PR DESCRIPTION
**Pull request by MSalopek**
## Description

Closes: #1535 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

File: x/ccv/provider/migrations/migrator.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	paramSpace     paramtypes.Subspace
+}
+
+// NewMigrator returns a new Migrator.
+func NewMigrator(providerKeeper providerkeeper.Keeper, paramSpace paramtypes.Subspace) Migrator {
+	return Migrator{providerKeeper: providerKeeper, paramSpace: paramSpace}
+}
+
+// Migrating consensus version 1 to 2 is a no-op.
+// <igrating from v1 -> v2 -> v3  will require multiple state breaking changes and migrations.
```
Comment (mpoke) [inline x/ccv/provider/migrations/migrator.go:23]: typo
```suggestion
// Migrating from v1 -> v2 -> v3  will require multiple state breaking changes and migrations.
```

File: x/ccv/provider/migrations/migrator.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// First run provider@v2.x.y in production to migrate from v1 -> v2. Then use provider@v3.x.y to migrate from v2 -> v3.
```
Comment (mpoke) [inline x/ccv/provider/migrations/migrator.go:24]:
```suggestion
// First run provider@v2.x.y in production to migrate from consensus version 1  to 2. 
// Then, in order to migrate to consensus version 3,  first upgrade to provider@v3.x.y.
```

Comment (mpoke) [reply x/ccv/provider/migrations/migrator.go:24]: Is it possible to upgrade to consensus version 3 directly from provider@v2.x.y?

File: x/ccv/provider/keeper/throttle_legacy.go [hunk 3]
```diff
@@ -4,52 +4,21 @@ import (
...
 // Note this type is copy/pasted from throttle v1 code.
 const (
 	slashPacketData byte = iota
 	vscMaturedPacketData
 )
 
-// GetAllThrottledPacketData returns all throttled packet data for a given consumer chain, only used for 2 -> 3 migration.
-// Note this method is adapted from throttle v1 code.
-func (k Keeper) GetAllThrottledPacketData(ctx sdktypes.Context, consumerChainID string) (
+// Deprecated: LegacyGetAllThrottledPacketData is deprecated is deprecated for ICS >= v4.0.0.
```
Comment (mpoke) [inline x/ccv/provider/keeper/throttle_legacy.go:19]:
```suggestion
// Deprecated: LegacyGetAllThrottledPacketData is deprecated for ICS >= v4.0.0.
```

File: x/ccv/provider/keeper/throttle_legacy.go [hunk 4]
```diff
@@ -4,52 +4,21 @@ import (
...
+// LegacyGetAllThrottledPacketData returns all throttled packet data that was queued on the provider for a given consumer chain.
+func (k Keeper) LegacyGetAllThrottledPacketData(ctx sdktypes.Context, consumerChainID string) (
```
Comment (mpoke) [inline x/ccv/provider/keeper/throttle_legacy.go:21]: `LegacyGetAllThrottledPacketData` would return empty lists for ICS >= v4, right?

**Review by mpoke — APPROVED** [review]
LGTM. See my comments below.

File: x/ccv/provider/migrations/migrator.go (hunk 2 above)
Comment (MSalopek) [reply x/ccv/provider/migrations/migrator.go:24]: Yes. `Migrate2to3` does that

File: x/ccv/provider/keeper/throttle_legacy.go (hunk 3 above)
Comment (MSalopek) [reply x/ccv/provider/keeper/throttle_legacy.go:19]: done

File: x/ccv/provider/keeper/throttle_legacy.go (hunk 4 above)
Comment (MSalopek) [reply x/ccv/provider/keeper/throttle_legacy.go:21]: Yes

**Review by bermuell — APPROVED** [review]

**Pull request merged**